tests/mpc.test.ts

```diff
@@ -40,38 +38,186 @@ test("max(3, 5) === 5", async () => {
     new EmpWasmBackend(),
   );
 
-  const aliceQueue = new AsyncQueue<Uint8Array>();
-  const bobQueue = new AsyncQueue<Uint8Array>();
+  const aqs = new AsyncQueueStore<Uint8Array>();
 
   const outputs = await Promise.all([
-    runSide(protocol, 'alice', { a: 3 }, aliceQueue, bobQueue),
-    runSide(protocol, 'bob', { b: 5 }, aliceQueue, bobQueue),
+    runParty(protocol, 'alice', { a: 3 }, aqs),
+    runParty(protocol, 'bob', { b: 5 }, aqs),
   ]);
 
   expect(outputs).to.deep.equal([{ main: 5 }, { main: 5 }]);
 });
 
-async function runSide(
+test("middle(8, 17, 5) == 8", async () => {
+  await summon.init();
+
+  const circuit = summon.compileBoolean('/src/main.ts', 8, {
+    '/src/main.ts': `
+      export default function main(
+        a: number,
+        b: number,
+        c: number,
+      ) {
+        const nums = [a, b, c];
+
+        const highest = a;
+        const secondHighest = 0;
+
+        for (let i = 1; i < nums.length; i++) {
+          if (nums[i] > highest) {
+            secondHighest = highest;
+            highest = nums[i];
+          } else if (nums[i] > secondHighest) {
+            secondHighest = nums[i];
+          }
+        }
+
+        return secondHighest;
+      }
+    `,
+  });
+
+  const mpcSettings = [
+    {
+      name: 'alice',
+      inputs: ['a'],
+      outputs: ['main'],
+    },
+    {
+      name: 'bob',
+      inputs: ['b'],
+      outputs: ['main'],
+    },
+    {
+      name: 'charlie',
+      inputs: ['c'],
+      outputs: ['main'],
+    },
+  ];
+
+  const protocol = new Protocol(
+    circuit,
+    mpcSettings,
+    new EmpWasmBackend(),
+  );
+
+  const aqs = new AsyncQueueStore<Uint8Array>();
+
+  const outputs = await Promise.all([
+    runParty(protocol, 'alice', { a: 8 }, aqs),
+    runParty(protocol, 'bob', { b: 17 }, aqs),
+    runParty(protocol, 'charlie', { c: 5 }, aqs),
+  ]);
+
+  expect(outputs).to.deep.equal([
+    { main: 8 },
+    { main: 8 },
+    { main: 8 },
+  ]);
+});
+
+// FIXME: this test is skipped
+// FIXME: use 5 bidders and auction house (which doesn't bid but observes)
+// NOTE:  this circuit also failed with consensus on the same bad outputs before
+//        the N parties upgrade, so there's something else going on:
+//        https://github.com/voltrevo/emp-wasm-backend/commit/be67477
+testOpt("vickrey(8, 17, 5) == [1, 8]", { skip: true }, async () => {
+  await summon.init();
+
+  const circuit = summon.compileBoolean('/src/main.ts', 8, {
+    '/src/main.ts': `
+      export default function main(
+        a: number,
+        b: number,
+        c: number,
+      ) {
+        const nums = [a, b, c];
+
+        const winner = 0;
+        const highest = a;
+        const secondHighest = 0;
```

Comment:
In JS const can't be re-assigned, I wonder why they can in Summon.

Reply:
Oh nice catch! Yeah that shouldn't happen 😅.

I think we're ignoring errors in summon-ts, task added here:
https://www.notion.so/pse-team/Fix-const-in-summon-ts-192d57e8dd7e801cbb7be0cc3a27c799?pvs=4